bundles/org.openhab.binding.goveelan/src/main/java/org/openhab/binding/goveelan/internal/GoveeLanHandler.java

```diff
@@ -0,0 +1,390 @@
+            // as we haven't received a packet we also don't know where it should have come from
+            // hence, we don't know which thing put offline.
+            // a way to monitor this would be to keep track in a list, which device answers we expect
+            // and supervise an expected answer within a given time but that will make the whole
+            // mechanism much more complicated and may be added in the future
+        } finally {
+            refreshJobRunning = false;
+        }
+    };
+
+    /**
+     * This thing related job <i>thingRefreshSender</i> triggers an update to the Govee device.
+     * The device sends it back to the common port and the response is
+     * then received by the common #refreshStatusReceiver
+     */
+    private final Runnable thingRefreshSender = () -> {
+        try {
+            triggerDeviceStatusRefresh();
+            if (!thing.getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/error.couldNotQueryDevice [\""
+                            + thing.getConfiguration().getProperties().get(GoveeLanConfiguration.IP_ADDRESS) + "\"]");
+        }
+    };
+
+    public GoveeLanHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        GoveeLanConfiguration goveeLanConfiguration = getConfigAs(GoveeLanConfiguration.class);
+
+        final Object ipAdress = getThing().getConfiguration().getProperties().get(GoveeLanConfiguration.IP_ADDRESS);
+        if (ipAdress != null) {
+            THING_HANDLERS.put(ipAdress.toString(), this);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/error.offline.ip-address.missing");
+        }
+        if (!THING_HANDLERS.isEmpty()) {
+            startRefreshStatusJob();
+        }
+
+        if (triggerStatusJob == null) {
+            LOGGER.debug("Starting refresh trigger job for thing {} ", thing.getLabel());
+
+            triggerStatusJob = scheduler.scheduleWithFixedDelay(thingRefreshSender, 100,
+                    goveeLanConfiguration.refreshInterval * 1000L, TimeUnit.MILLISECONDS);
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+    }
+
+    /**
+     * Stop the Refresh Status Job, so the same socket can be used for something else (like discovery)
+     */
+    public static void stopRefreshStatusJob() {
+        if (refreshStatusJob != null) {
+            refreshStatusJob.cancel(true);
+        }
+        refreshStatusJob = null;
+        refreshJobRunning = false;
+    }
+
+    /**
+     * (re)start the refresh status job
+     */
+    public static void startRefreshStatusJob() {
+        if (refreshStatusJob == null) {
+            refreshStatusJob = ThreadPoolManager.getScheduledPool("thingHandler")
```

Comment:
I think some kind of identification would be useful here, perhaps "binding-govee". I'm not sure if there are any strict conventions, looking at the codebase it doesn't seem so.